File: study_ws/src/cpp_pkg/src/radio.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class radioNode : public rclcpp::Node 
{
public:
    radioNode() : Node("radio") 
    {
        subscriber_ = this->create_subscription<std_msgs::msg::String>("news", 10, 
                            std::bind(&radioNode::topic_callback, this, std::placeholders::_1));
        RCLCPP_INFO(this->get_logger(), "Capturing Buffer...");
    }

private:

    void topic_callback(const std_msgs::msg::String::SharedPtr msg) 
    {
        RCLCPP_INFO(this->get_logger(), "I heard: '%s'", msg->data.c_str());
    }
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscriber_;


};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<radioNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: study_ws/src/py_pkg/py_pkg/radio.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from std_msgs.msg import String

class radio(Node): 
    def __init__(self):
        super().__init__("radio") 
        #self.counter = 0
        self.subscriber_ = self.create_subscription(String, "news", self.listener_callback, 10)
        self.get_logger().info("Listening..")

    def listener_callback(self, msg):
        #self.counter += 1
        self.get_logger().info("I heard: " + msg.data)

def main(args=None):
    rclpy.init(args=args)
    node = radio() 
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()
